arch/sim/src/sim/up_initialstate.c

```diff
@@ -67,6 +67,6 @@
 void up_initial_state(struct tcb_s *tcb)
 {
   memset(&tcb->xcp, 0, sizeof(struct xcptcontext));
```

Comment:
i don't understand your suggestion. can you explain?

Reply:
Like the below dummy field:
struct xcptcontext
{
   void *sigdeliver; /* Actual type is sig_deliver_t */
   xcpt_reg_t dummy;
   xcpt_reg_t regs[XCPTCONTEXT_REGS];
};

Reply:
this change is about the actual stack alignment. that is, the value of RSP register.
not about the context structure.

Reply:
My previous comment is wrong, but all places(up_use_stack/up_stack_frame) which modify adj_stack_ptr already align this pointer to 16B boundary, the extra subtraction just make the situation even worse, I can't understand why this change fix your problem.

Reply:
the convention is, align _before_ a function call.
that is, from POV of the function being called, it will be aligned after pushing RBP.
note that up_longjmp jumps into the function directly. (without pushing the return address)
RSP at that point should not be aligned.